File: picocli-examples/src/main/kotlin/picocli/examples/kotlin/subcommands/TopCmd.kt
package picocli.examples.kotlin.subcommands

import picocli.CommandLine
import picocli.CommandLine.*
import picocli.CommandLine.Model.CommandSpec

@Command(name = "top", version = ["Kotlin picocli demo v3.8.1"],
        mixinStandardHelpOptions = true,
        description = ["@|bold Kotlin|@ @|underline picocli|@ example"],
        subcommands = [SubCmd::class])
class TopCmd : Runnable {
    @Spec
    var spec: CommandSpec? = null

    override fun run() {
        throw ParameterException(spec?.commandLine(), "Specify a subcommand")
    }
}
fun main(args: Array<String>) = System.exit(CommandLine(TopCmd()).execute(*args))

File: picocli-examples/src/main/scala/picocli/examples/scala/subcommands/SubCmdsViaClasses/TopCmd.scala
package picocli.examples.scala.subcommands.SubCmdsViaClasses

import picocli.CommandLine
import picocli.CommandLine.{Command, HelpCommand}

@Command(name = "ISOCodeResolve", version = Array("4.1.4"),
  mixinStandardHelpOptions = true,
  subcommands = Array(classOf[SubCmd1], classOf[SubCmd2], classOf[HelpCommand]),
  description = Array("Resolve ISO country codes (ISO-3166-1) or language codes (ISO 639-1 or -2)"))
class TopCmd extends Runnable {
  @CommandLine.Spec
  val spec: CommandLine.Model.CommandSpec = null

  def run(): Unit = {
    throw new CommandLine.ParameterException(spec.commandLine(), "Specify a subcommand")
  }
}

object TopCmd {
  def main(args: Array[String]): Unit = {
    System.exit(new CommandLine(new TopCmd()).execute(args: _*))
  }
}
